### PR TITLE
Separate CI jobs

### PR DESCRIPTION
**Pull request by szepeviktor**
Much better usability.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,52 +1,42 @@
...
-# 1. DOWNGRADE_CODE="true"
-# 2. DOWNGRADE_CODE="false"
-#
-# When DOWNGRADE_CODE is "false", it is the standard testing pipeline:
-# Execute PHPCS, PHPUnit and PHPStan to check there are no issues, and generate report.
-#
-# When DOWNGRADE_CODE is "true", it tests the conversion of the PHP code version 7.4 to 7.1:
-# - Run Rector to downgrade the code
-# - Run PHPStan. If it exits with no errors, then downgrading the code works well.
-########################################################################
```
Comment (leoloso) [inline .travis.yml:19]: Please don't remove the documentation. I want to explain the objective of the PHP 7.1 job (previously through env var `DOWNGRADE`CODE`)

File: .travis.yml [hunk 2]
```diff
@@ -1,52 +1,42 @@
...
+        - vendor/bin/phpstan analyse src/
+      after_script: skip
+
+    - name: "PHP 7.1"
+      install:
+        - travis_retry composer update --no-interaction --prefer-dist
+        - vendor/bin/rector process
+        - phpenv local 7.1
+        - php --version
+        - vendor/bin/phpstan analyse src/
```
Comment (leoloso) [inline .travis.yml:29]: Use configuration: `vendor/bin/phpstan analyse -c phpstan.neon.dist src`

File: .travis.yml [hunk 3]
```diff
@@ -1,52 +1,42 @@
...
-  # Tests only on PHP 7.4
-  - if [[ "$DOWNGRADE_CODE" != "true" ]]; then vendor/bin/phpcs -n --standard=psr2 src/; fi
-  - if [[ "$DOWNGRADE_CODE" != "true" ]]; then vendor/bin/phpunit --coverage-text --coverage-clover=coverage.clover; fi
-  # Switch to PHP 7.1 to test if downgraded code works, then up again to 7.4
-  - if [[ "$DOWNGRADE_CODE" == "true" ]]; then phpenv local 7.1; fi
-  - if [[ "$DOWNGRADE_CODE" == "true" ]]; then php --version; fi
-  - vendor/bin/phpstan analyse -c phpstan.neon.dist src
-  - if [[ "$DOWNGRADE_CODE" == "true" ]]; then phpenv local 7.4; fi
-  - if [[ "$DOWNGRADE_CODE" == "true" ]]; then php --version; fi
+  - vendor/bin/phpunit --coverage-text --coverage-clover=coverage.clover
```
Comment (leoloso) [inline .travis.yml:36]: This must not run on the PHP 7.1 job. It will always fail, because I'm using PHPUnit 9.3

File: .travis.yml (hunk 2 above)
Comment (szepeviktor) [reply .travis.yml:29]: okay

File: .travis.yml (hunk 3 above)
Comment (szepeviktor) [reply .travis.yml:36]: This does not run on PHP 7.1
There are 1+3 jobs.

Comment (leoloso) [reply .travis.yml:36]: It did run, that's why it failed

Comment (leoloso) [reply .travis.yml:36]: Now it's working :)

File: .travis.yml (hunk 1 above)
Comment (szepeviktor) [reply .travis.yml]: Done.

**Comment by szepeviktor** [comment]
This is insance!

kép

I see only 3 jobs.
https://travis-ci.com/github/GraphQLAPI/graphql-api-for-wp/builds/185916201

**Comment by leoloso** [comment]
You mean `phpunit` is missing, right?

**Comment by szepeviktor** [comment]
Yes, the default job does not start.

**Comment by leoloso** [comment]
Could it be happening because of `after_script: skip` in all jobs? Shouldn't it be added only to the PHP 7.1 job?

In the [previous build](https://travis-ci.com/github/GraphQLAPI/graphql-api-for-wp/jobs/388951189), it reads: `Skipping the after_script step, as specified in the configuration.`

**Comment by leoloso** [comment]
I know `phpunit` is in `script`, not in `after_script`, but I wonder

**Comment by szepeviktor** [comment]
When I reduced it to 1 jobs, it couldn't start:

https://travis-ci.com/github/GraphQLAPI/graphql-api-for-wp/jobs/388954069

kép

**Comment by szepeviktor** [comment]
> Could it be happening because of `after_script: skip` in all jobs? Shouldn't it be added only to the PHP 7.1 job?

This ensures that coverage data is uploaded only in the default jobs.

**Comment by leoloso** [comment]
Otherwise, what about making `PHPUnit` another job? Why 1+3 and not 4?

**Comment by szepeviktor** [comment]
> Otherwise, what about making `PHPUnit` another job? Why 1+3 and not 4?

Yes that is a work-around, leaving normality behind :)

**Comment by leoloso** [comment]
If it works normally, great! If not, I leave the decision to you :)

**Comment by szepeviktor** [comment]
I've sent an email to TravisCI support.

**Comment by szepeviktor** [comment]
We have a problem where we do not look currently.
A Hidden One!

**Comment by leoloso** [comment]
Why doesn't it work with bionic?

Btw, I was using xenial before, because it has PHP 7.1 preinstalled. But using bionic also seemed to work about that...

**Comment by szepeviktor** [comment]
Defeat, Leo, defeat!

kép

**Comment by szepeviktor** [comment]
Here is it!!!

https://github.com/travis-ci/travis-ci/issues/4681#issuecomment-304919767

**Comment by leoloso** [comment]
Thanks @szepeviktor! Are you happy with this config now? Should I merge it? :)

**Comment by szepeviktor** [comment]
Yes please.

**Comment by leoloso** [comment]
Thank you :)

**Comment by szepeviktor** [comment]
You're welcome.

**Pull request merged**